x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/utils/create_detection_rule_from_vulnerability.ts

```diff
@@ -139,7 +140,7 @@ export const createDetectionRuleFromVulnerabilityFinding = async (
         group_by: [ALERT_SUPPRESSION_FIELD],
         missing_fields_strategy: AlertSuppressionMissingFieldsStrategy.Suppress,
       },
-      index: [VULNERABILITIES_INDEX_PATTERN],
+      index: [VULNERABILITIES_INDEX_PATTERN, CDR_LATEST_THIRD_PARTY_VULNERABILITIES_INDEX_PATTERN],
```

Comment:
I referenced the latest index here (`CDR_LATEST_THIRD_PARTY_VULNERABILITIES_INDEX_PATTERN`), but there may be cases where the vulnerability hasn't yet appeared in the latest index and is only present in the source index. I'm unsure if we can reference the source index in such cases within the CDR.

Reply:
I like that we are trying to deliver more than we intended (as in make detection rules work for 3rd party data) but tbh I'd follow the initial plan and implement the detection rule support as a separate step exactly for this reason: I don't think having different logic for native and 3rd party (for native create alerts based on new events in the data stream and for 3rd party based on the latest index) is a good idea. There were reasons why we implemented the detection rules based on the data streams and if these reasons don't hold true anymore, we should update them consistently. Otherwise, it isn't very clear. I would leave detection rule support as a separate work item